ImageGallery/src/org/sleuthkit/autopsy/imagegallery/ImageGalleryModule.java

```diff
@@ -378,6 +378,7 @@ public void propertyChange(PropertyChangeEvent event) {
                              * database.
                              */
                             if (controller.isListeningEnabled()) {
+                                controller.getGroupManager().resetLastUpdatedPathGroup();
```

Comment:
@bcarrier and @raman-bt, the need for this call concerns me a bit. As I understand it, we need some way to mop up after the last path is completed. That means that this inner event handler class in the ImageGalleryModule class is tightly coupled to implementation details inside of the GroupManager. At the very least we need to comment this. But I have a growing conviction that instead of having event handlers in ImageGalleryModule that try to manipulate the rest of the system, the interested components should have their own event handlers. I think the code would be simpler and sounder then.